app.r

```diff
@@ -14,31 +15,149 @@ ui <- fluidPage(
         tags$li(a("Work", href="#work")), 
         tags$li(a("Community", href="#community")), 
-        tags$li(a("Methodology", href="#methodology")), 
+        tags$li(a("Methodology", href="#methodology"))
       )
     ),
     mainPanel(
       fluidRow(h4(id="overview","Overview")),
-      fluidRow(h5("This year, over 100,000 developers told us how they learn, build their careers, which tools they’re using, and what they want in a job.")),
+      fluidRow(h5("This year, over 100,000 developers told us how they learn, build their careers, which tools they're using, and what they want in a job.")),
       fluidRow(
+        plotOutput("occurences_plot"),
         plotOutput("student_occurences_plot"),
         plotOutput("prof_occurences_plot")    
-      )),
+      ),
       fluidRow(h4(id="developer_profile","Developer Profile")),
-      fluidRow(h4(id="technology","Technology")),
-      fluidRow(h4(id="work","Work")),
-      fluidRow(h4(id="community","Community")),
-      fluidRow(h4(id="methodology","Methodology"))
+      fluidRow(h5("Who codes? More people in more places than ever before.")),
+      fluidRow(
+        plotOutput("profDeveloperYears_plot"),
+        plotOutput("ethnicity_occurences_plot"),
+        plotOutput("ethnicity_prof_occurences_plot"),
+        plotOutput("ethnicity_student_occurences_plot")
+    ),
+    fluidRow(h4(id="technology","Technology")),
+    fluidRow(h4(id="work","Work")),
+    fluidRow(h4(id="community","Community")),
+    fluidRow(h4(id="methodology","Methodology"))
   )
-)
+))
 
 
 
 server <- function(input, output){
   surveyResultsPublic <- read.csv("/Users/paulkelly/Dropbox/lecturing/GroupProject/stackoverflow_insights/datasets/survey_results_public.csv")
   
+  #Gender All Respondents
+  
+  genders <- subset.data.frame(surveyResultsPublic, select = c("Gender"))
+  
+  genders <- na.omit(genders)
+  
+  View(genders)
+  
+  count_all_genders <- separate(data = genders, col = Gender, into = c("Var1", "Var2", "Var3", "Var4", "Var5"), sep = ";")
+  
+  View(count_all_genders)
+  
+  occurences <- as.data.frame(table(unlist(count_all_genders)))
+  
+  View(occurences)
+  
+  str(occurences)
+  
+  occurences$Var1 <- as.factor(occurences$Var1)
+  
+  #combines Female gender
+  occurences[1, ] <- occurences[1, ] + occurences[5, ]
+  occurences[1, 1] <- "Female"
+  occurences <- occurences[-c(5),]
+  
+  #combines Transgender 
+  occurences [4, ] <- occurences[4, ] + occurences[8, ]
+  occurences[4, 1] <- "Transgender"
+  occurences <- occurences[-c(8),]
+  
+  #combines Other
+  occurences [3, ] <- occurences[3, ] + occurences[7, ]
+  occurences[3, 1] <- "Other"
+  occurences <- occurences[-c(7),]
+  
+  #combines Gender Non-Conforming
+  occurences [2, ] <- occurences[2, ] + occurences[5, ]
+  occurences[2, 1] <- "Gender non-conforming"
+  occurences <- occurences[-c(5),]
+  
+  
+  occurences$Percentage <- occurences$Freq / sum(occurences$Freq) * 100
+  
+  output$occurences_plot<renderPlot({
```

Comment:
Vinesh, this line is causing an error, should be <-renderPlot instead of <renderPlot